assets/js/components/OverlayNotification/AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification.test.js

```diff
@@ -0,0 +1,465 @@
+			{
+				registry,
+				features: [ 'ga4AdSenseIntegration' ],
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
+			}
+		);
+		expect( container ).not.toHaveTextContent(
+			'Data is now available for the pages that earn the most AdSense revenue'
+		);
+	} );
+
+	it( 'does not render in "view only" dashboard without AdSense access', () => {
+		provideUserAuthentication( registry, { authenticated: false } );
+		registry
+			.dispatch( CORE_USER )
+			.receiveGetCapabilities( capabilitiesAdSenseNoAccess.permissions );
+		const { container } = render(
+			<AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification />,
+			{
+				registry,
+				features: [ 'ga4AdSenseIntegration' ],
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
+			}
+		);
+		expect( container ).not.toHaveTextContent(
+			'Data is now available for the pages that earn the most AdSense revenue'
+		);
+	} );
+
+	it( 'renders in "view only" dashboard with Analytics and AdSense access', () => {
+		provideUserAuthentication( registry, { authenticated: false } );
+		registry
+			.dispatch( CORE_USER )
+			.receiveGetCapabilities( capabilities.permissions );
+		provideAnalytics4MockReport( registry, reportOptions );
+
+		const { container } = render(
+			<AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification />,
+			{
+				registry,
+				features: [ 'ga4AdSenseIntegration' ],
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
+			}
+		);
+		expect( container ).toHaveTextContent(
+			'Data is now available for the pages that earn the most AdSense revenue'
+		);
+	} );
+
+	it( 'renders `Show me` and `Maybe later` buttons`', () => {
+		provideAnalytics4MockReport( registry, reportOptions );
+
+		const { container } = render(
+			<AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification />,
+			{
+				registry,
+				features: [ 'ga4AdSenseIntegration' ],
+			}
+		);
+
+		expect( container ).toHaveTextContent( 'Show me' );
+		expect( container ).toHaveTextContent( 'Maybe later' );
+	} );
+
+	it( 'clicking the `Show me` button dismisses the notification', async () => {
+		provideAnalytics4MockReport( registry, reportOptions );
+		fetchMock.getOnce( fetchGetDismissedItemsRegExp, { body: [] } );
+		fetchMock.postOnce( fetchDismissItemRegExp, {
+			body: [ ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION ],
+		} );
+
+		const { container, getByRole, rerender, waitForRegistry } = render(
+			<AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification />,
+			{
+				registry,
+				features: [ 'ga4AdSenseIntegration' ],
+			}
+		);
+
+		await waitForRegistry();
+
+		expect( container ).toHaveTextContent(
+			'Data is now available for the pages that earn the most AdSense revenue'
+		);
+
+		act( () => {
+			fireEvent.click( getByRole( 'button', { name: /Show me/i } ) );
+		} );
+
+		rerender();
+
+		expect( container ).not.toHaveTextContent(
+			'Data is now available for the pages that earn the most AdSense revenue'
```

Comment:
Hm, currently there is no dismissal dispatched when `Show me` CTA is clicked. It might be that this checks runs early before the component finishes re-rendering. Maybe we can include the check if dismissal action have been fetch, or include additional timeout/waitForRegistry after `rerender` is called?